cmake/extensions.cmake

```diff
@@ -400,6 +400,11 @@ function(zephyr_library_compile_options item)
   string(MD5 uniqueness ${item})
   set(lib_name options_interface_lib_${uniqueness})
 
+  if (TARGET ${lib_name})
+    message(WARNING "${item} already added, ignoring duplicate")
```

Comment:
We should drop the warning to align with CMake's silent de-duplication.

Reply:
Are you sure? I thought the silent de-duplication was a bug you just discovered. Is it really a feature?

Reply:
It is CMake behaviour that we want to align with to be consistent with CMake.

Reply:
Generally we want to keep the logs quiet. It's best to prefer errors and avoid warnings. In this case we can't use an error as that would be inconsistent with CMake.

Reply:
@marc-hb can you please address the above?

Reply:
I will remove the warning.

The entire and unique purpose of this PR is to restore the same behaviour than CMake, so no question about that. I would still appreciate more clarity on that behaviour though; I mean I would still like to know whether that behaviour is a CMake _feature_ or a CMake _bug_. * If * adding the same option twice is considered a usage error, then users would rather be told about their error than enjoy the consistency of wasting as much time as they would have with CMake. Now I don't work in any Zephyr support role right now so this is admittedly just for my personal education.

Reply:
please submit the change ASAP, this PR is resolved a P1 bug, we need to clear the P1 bug list ASAP, otherwise we might resort to reverting the change introducing this issue.

Reply:
I will submit the change today.

Sorry I had no idea issue #15093 was now high priority, this is complete news to me. Github doesn't notify label changes and #15093 is not even assigned to me, it's still assigned to @SebastianBoe . I asked in #15093 about its priority but no one gave any assessment.
Please make more github noise next time.

Reply:
It's a CMake feature.

They have explicitly written code to de-duplicate the flags.